src/hooks/use-debounce.ts

```diff
@@ -0,0 +1,17 @@
+import { useState, useEffect } from 'react';
+
+export function useDebounce(value: string, delay: number) {
+  const [debouncedValue, setDebouncedValue] = useState(value);
+
+  useEffect(() => {
+    const handler = setTimeout(() => {
+      setDebouncedValue(value);
+    }, delay);
+
+    return () => {
+      clearTimeout(handler);
+    };
```

Comment:
this is not particularly the hanlder but the timeout id
```suggestion
    const timeoutId = setTimeout(() => {
      setDebouncedValue(value);
    }, delay);

    return () => {
      clearTimeout(timeoutId);
    };
```

Reply:
Thanks! You're right, I'll rename it.